python/tests/test_tables.py

```diff
@@ -921,6 +921,41 @@ def test_pickle_packing(self):
         self.assertEqual(data, unpickled)
 
 
+class TestNumpyInt32Conversion(unittest.TestCase):
+    """
+    Tests the function that converts integer arrays to np.int32 for input to functions.
+    """
+    def test_arrays(self):
+        # Simple array
+        target = np.array([0, 1], dtype=np.int32)
+        for identical_array in [[0, 1], (0, 1), np.array([0, 1]), target]:
+            converted = tskit.tables.to_np_int32(identical_array)
+            # Use pickle to test exact equality including dtype
+            self.assertEqual(pickle.dumps(converted), pickle.dumps(target))
+        # Nested array
+        target = np.array([[0, 1], [2, 3]], dtype=np.int32)
+        for identical_array in [[[0, 1], [2, 3]], np.array([[0, 1], [2, 3]]), target]:
+            converted = tskit.tables.to_np_int32(identical_array)
+            self.assertEqual(pickle.dumps(converted), pickle.dumps(target))
+
+    def test_empty_arrays(self):
+        target = np.array([], dtype=np.int32)
+        converted = tskit.tables.to_np_int32([])
+        self.assertEqual(pickle.dumps(converted), pickle.dumps(target))
+        target = np.array([[]], dtype=np.int32)
+        converted = tskit.tables.to_np_int32([[]])
+        self.assertEqual(pickle.dumps(converted), pickle.dumps(target))
+
+    def test_bad_types(self):
+        for bad_type in [[0.1], ['string'], {}, [{}], np.array([0, 1], dtype=np.float)]:
+            self.assertRaises(TypeError, tskit.tables.to_np_int32, bad_type)
+
+    def test_overflow(self):
+        for bad_node in [np.iinfo(np.int32).min-1, np.iinfo(np.int32).max+1]:
+            self.assertRaises(
+                OverflowError, tskit.tables.to_np_int32, np.array([0, bad_node]))
+
```

Comment:
Looks good. I think we just need one more test, where we try out the values what are just inside the allowable range to make sure the boundaries are handled correctly.

Reply:
Good point

Reply:
Done